app.js

```diff
@@ -149,7 +149,8 @@ app.use(['/apply', '/download'], async (req, res, next) => {
     if (!req.session.questionnaireId) {
         // no: set it and redirect.
         try {
-            const response = await qService.createQuestionnaire();
+            const userId = req.session.userId ? req.session.userId : undefined;
+            const response = await qService.createQuestionnaire(userId);
```

Comment:
````js
const response = await qService.createQuestionnaire({
    userId: req.session.userId
});
````

this function signature allow function overloading (sort of) - good pattern to have. And avoids the need to create a `userId const` due to the `undefined` check being done in `questionnaire-service.js:23`